lib/dialects/postgres/query-generator.js

```diff
@@ -805,8 +806,10 @@ var QueryGenerator = {
   },
 
   pgEnumAdd: function(tableName, attr, value, options) {
-    var enumName = this.pgEnumName(tableName, attr)
-      , sql = 'ALTER TYPE ' + enumName + ' ADD VALUE ' + this.escape(value);
+    var enumName = this.pgEnumName(tableName, attr);
+    var sql = semver.gte(this.sequelize.options.databaseVersion, '9.3.0')
+        ? 'ALTER TYPE ' + enumName + ' ADD VALUE IF NOT EXISTS ' + this.escape(value)
+        : 'ALTER TYPE ' + enumName + ' ADD VALUE ' + this.escape(value);
```

Comment:
```js
sql = 'ALTER TYPE ' + enumName + ' ADD VALUE';

if (semver.gte(this.sequelize.options.databaseVersion, '9.3.0')) {
  sql += 'IF NOT EXISTS';
}
sql +=  this.escape(value)
```

Nitpicking now, but this is slightly more DRY, and less prone to errors where you refactor one case but not the other :)